streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -5715,37 +5716,73 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
   }
 
   private[zio] class Rechunker[A](n: Int) {
-    private var buffer: ChunkBuilder[A] = if (n > 1) ChunkBuilder.make(n) else null
-    private var pos: Int                = 0
+    private var buffer: Chunk[A]              = if (n > 1) Chunk.empty[A] else null
+    private var chunkBuilder: ChunkBuilder[A] = null
+    private var pos: Int                      = 0
 
     def isEmpty: Boolean = pos == 0
 
     final def rechunk(
       chunk: Chunk[A]
     )(implicit trace: Trace): ZChannel[Any, ZNothing, Any, Any, ZNothing, Chunk[A], Any] = {
-      val len = chunk.size
-      if (len == 0) {
+      val chunkSize = chunk.size
+      if (chunkSize == 0) {
         null
-      } else if (isEmpty && len == n) {
+      } else if (isEmpty && chunkSize == n) {
         ZChannel.write(chunk)
       } else if (n == 1) {
-        rechunk1(chunk, len)
-      } else {
+        rechunk1(chunk, chunkSize)
+      } else if (chunkSize < 22) {
```

Comment:
I'm a bit surprised on the choice of 22 here. I would have imagined that if there was a cut-off for observing a difference it would be one of [these values](https://github.com/zio/zio/blob/69b2ecf95fa0e9ea3e59f635bbfd6cf839a4c967/core/shared/src/main/scala/zio/Chunk.scala#L1427-L1443)

Reply:
I was surprised as well.